cyberdrop_dl/crawlers/xgroovy.py

```diff
@@ -0,0 +1,133 @@
+from __future__ import annotations
+
+import json
+from enum import StrEnum
+from typing import TYPE_CHECKING, ClassVar, NamedTuple
+
+from cyberdrop_dl.crawlers.crawler import Crawler, SupportedPaths
+from cyberdrop_dl.data_structures.url_objects import AbsoluteHttpURL
+from cyberdrop_dl.utils import css, open_graph
+from cyberdrop_dl.utils.utilities import error_handling_wrapper
+
+if TYPE_CHECKING:
+    from bs4 import BeautifulSoup
+
+    from cyberdrop_dl.data_structures.url_objects import ScrapeItem
+
+
+PRIMARY_URL = AbsoluteHttpURL("https://xgroovy.com")
+PICTURES_DOMAIN = "photos.xgroovy.com"
+
+
+class Selectors:
+    VIDEO = "video#main_video"
+    GIF = "div.gif-video-wrapper > video"
+    UPLOAD_DATE = "script:-soup-contains('uploadDate')"
+    PORNSTAR_NAME = "h2.object-title"
+    SEARCH_VIDEOS = "div.list-videos a.popito"
+    NEXT_PAGE = "div.pagination-holder li.next > a"
+
+
+_SELECTORS = Selectors()
+
+
+class Format(NamedTuple):
+    resolution: str
+    link_str: str
+
+
+class CollectionType(StrEnum):
+    CATEGORIES = "categories"
+    CHANNELS = "channels"
+    PORNSTARS = "pornstars"
+    SEARCH = "search"
+    TAG = "tag"
+
+
+class XGroovyCrawler(Crawler):
+    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {
+        "Video": ("/<category>/videos/<video_id>/...", "/videos/<video_id>/..."),
+        "Gif": ("/<category>/gifs/<gif_id>/...", "/gifs/<gif_id>/..."),
+        "Search": ("/<category>/search/...", "/search/..."),
+        "Pornstar": ("/<category>/pornstars/<pornstar_id>/...", "/pornstars/<pornstar_id>/..."),
+        "Tag": ("/<category>/tags/...", "/tags/..."),
+        "Channel": ("/<category>/channels/...", "/channels/..."),
+    }
```

Comment:
I don't think these paths are correct. All of the have `<category>`

Please also add a commas at the end of each tuple to force ruff to format them vertically:


```suggestion
    SUPPORTED_PATHS: ClassVar[SupportedPaths] = {
        "Video": ("/<category>/videos/<video_id>/...", "/videos/<video_id>/...",),
        "Gif": ("/<category>/gifs/<gif_id>/...", "/gifs/<gif_id>/...",),
        "Search": ("/<category>/search/...", "/search/...",),
        "Pornstar": ("/<category>/pornstars/<pornstar_id>/...", "/pornstars/<pornstar_id>/...",),
        "Tag": ("/<category>/tags/...", "/tags/...",),
        "Channel": ("/<category>/channels/...", "/channels/...",),
    }
```

Reply:
The paths are correct and working. The site also has the categories /shemale and /gay.
Ruff complains about the ending commas: 
`Trailing comma prohibitedRuff[COM819](https://docs.astral.sh/ruff/rules/prohibited-trailing-comma)`